awx/api/serializers.py

```diff
@@ -2297,20 +2305,41 @@ def build_field(self, field_name, info, model_class, nested_depth):
         return super(V1JobOptionsSerializer, self).build_field(field_name, info, model_class, nested_depth)
 
 
+@six.add_metaclass(BaseSerializerMetaclass)
+class LegacyCredentialFields(BaseSerializer):
```

Comment:
I decided to implement this as a _separate_ mixin because the v1 fields (`cloud_credential` and `network_credential`) have been deprecated for an entire release cycle and will be removed _before_ these go away.

Reply:
I don't see `LegacyCredentialFields` as a parent class for any other serializer class (a simple ctrl+F on this PR). So I would assume something went wrong with applying it?

Reply:
It's not used as a base class, but is instead applied here:  https://github.com/ansible/awx/pull/595/files#diff-a81324c523b41de7296fdd5ff9063d10R2342

Reply:
you're including `*` in the fields, and the model here is Credential. so you have a dictionary with the BaseSerializer fields built from the Credential model, and then you're combing within Job / JobTemplate fields, with the credential base fields taking precedence over the Job-like thing base fields (`fields.update(LegacyCredentialFields().get_fields())`).

This probably won't ever cause any problems, sure, because credentials have the same name/description base stuff as Jobs, but if someone tried to modify a tiny detail of one of those fields in JobOptions, they would be in for a world of hurt.

Reply:
Yea.  I went through several versions of trying to make this work with our fields metaclass in `/api/v1/`, and this was the _least_ painful thing I came up with.  It also silos these off so that removal is fairly simple in the future once we pass the deprecation period.